client/my-sites/checkout/composite-checkout/composite-checkout.js

```diff
@@ -622,11 +630,21 @@ function useRedirectIfCartEmpty( items, redirectUrl, isLoading ) {
 	useEffect( () => {
 		if ( prevItemsLength > 0 && items.length === 0 ) {
 			debug( 'cart has become empty; redirecting...' );
+			if ( isLoggedOutCart ) {
+				window.localStorage.removeItem( 'siteParams' );
+				window.location = redirectUrl;
```

Comment:
Just out of curiosity, what's the reason to use `window.location` here instead of the `page.redirect()` below? Does the latter not work correctly in some circumstances?

Reply:
Using `page.redirect()` takes to the login page as I suspect the cookies are not being detected. Doing a page reload with window.location fixes the issue.

Reply:
Sounds good. Could you put a comment in the code to explain this briefly? Otherwise I'm sure that someone (maybe me!) will break this in the future thinking that there's no difference.

Reply:
Makes sense, I've added the comment.